docs/proposals/resource-deletion-with-approval.md

```diff
@@ -0,0 +1,157 @@
+
+In order to improve the situation, we propose to introduce `confirm` option for Prune sync option. When `confirm` option is set, Argo CD should pause the sync operation
+**before deleting any app resources** and wait for the user to confirm the deletion. The confirmation can be done in a very friendly way using Argo CD UI, CLI or API.
+
+
+### Introduce `confirm` option for Delete sync option.
+
+Similarly to `Prune` sync option we need to introduce `confirm` value for `Delete` sync option: `argocd.argoproj.io/sync-options: Delete=confirm`. The `confirm` option
+should pause the sync operation **before deleting any app resources** and wait for the user to confirm the deletion. The confirmation can be done in a very friendly way
+using Argo CD UI, CLI or API.
+
+
+### Friendly prunning/deletion manual approval
+
+Since we know Argo CD is often used to implement fully automated developer workflows that include resource deletion, the
+deletion approval process should be as painless as possible. This way, platform administrators can instruct end users to
+apply the new prune/delete option to resources that require special care without significantly disturbing the developer
+experience.
+
+In both cases where Argo CD requires manual approval, the user should be able to approve the deletion using Argo CD UI,
+CLI, or API. The approval process should be as simple as possible and should not require the user to understand the
+internals of Argo CD.
+
+#### New `requiresDeletionApproval` resource field in application status
+
+A new field `requiresDeletionApproval` should be added to the `status.resources` list items. The field should be set to `true` when the resource deletion approval is required.
+
+```yaml
+  - health:
+      status: Healthy
+    kind: Service
+    name: guestbook-ui
+    namespace: default
+    status: OutOfSync
+    version: v1
+    requiresPruning: true
+    requiresDeletionApproval: true # new field that indicates that deletion approval is required
+```
+
+The Argo CD UI, CLI should visualize the `requiresDeletionApproval` field so that the user can easily discover which resources require manual approval.
+
+#### Approve deletion resource action
+
+The Argo CD UI, CLI should bundle the `Approve Deletion` [resource action](https://argo-cd.readthedocs.io/en/stable/operator-manual/resource_actions/)
+that would allow the user to approve the deletion. The action should patch the resource with the `argocd.argoproj.io/deletion-approved: true` annotation.
+Once annotation is applied the Argo CD should proceed with the deletion.
+
+The main reason to use the action is that we can reuse existing [RBAC](https://argo-cd.readthedocs.io/en/stable/operator-manual/rbac/) to control who can approve the deletion.
+
+#### UI Convinience to approve all resources
+
+The Argo CD UI should provide a convinient way to approve all resources that require manual approval. This way the user can approve all resources with a single click. 
```

Comment:
Does this mean all resources within a given app? And maybe for the CLI, even if I'm deleting multiple apps in one command?

Reply:
Resources of a given app. Added clarification about both UI and CLI